### PR TITLE
CMake: Add Google Benchmark (with FetchContent)

### PR DESCRIPTION
**Pull request by meastp**
Adds google benchmark to dependencies, and opencensus' benchmarks to the cmake build

### CODE REVIEW

**Comment by g-easy** [comment]
Please fix the cmake-on-linux break on Travis: https://travis-ci.org/census-instrumentation/opencensus-cpp/jobs/506125611

Regarding format, please try out https://github.com/census-instrumentation/opencensus-cpp/pull/299

**Comment by meastp** [comment]
@g-easy Unfortunately, the docker container did not work fully. All the sed commands fail with `sed: cannot rename ./tools/sedMOsykq: No such file or directory`, I'm guessing because of limitations in mounting/binding volumes (sed -i creates a temp file). When I comment out the sed commands in format.sh, everything works... Are there other options for replacing text in files...?

**Comment by meastp** [comment]
I managed to run format.sh (with some modifications, see #310).

**Comment by g-easy** [comment]
use explicit commit for abseil -> is this fixed by #307?

File: cmake/OpenCensusHelpers.cmake [hunk 1]
```diff
@@ -38,6 +38,18 @@ function(opencensus_test NAME SRC)
   endif()
 endfunction()
 
+# Helper function like bazel's cc_test. Usage:
+#
+# opencensus_benchmark(trace_some_test internal/some_test.cc dep1 dep2...)
+function(opencensus_benchmark NAME SRC)
+  # if(BUILD_TESTING)
```
Comment (g-easy) [inline cmake/OpenCensusHelpers.cmake]: IMO benchmarks should only be built if BUILD_TESTING is enabled. Please uncomment.

File: cmake/OpenCensusHelpers.cmake [hunk 2]
```diff
@@ -38,6 +38,18 @@ function(opencensus_test NAME SRC)
   endif()
 endfunction()
 
+# Helper function like bazel's cc_test. Usage:
+#
+# opencensus_benchmark(trace_some_test internal/some_test.cc dep1 dep2...)
```
Comment (g-easy) [inline cmake/OpenCensusHelpers.cmake]: s/_test/_benchmark/

File: cmake/OpenCensusHelpers.cmake [hunk 3]
```diff
@@ -38,6 +38,18 @@ function(opencensus_test NAME SRC)
...
+function(opencensus_benchmark NAME SRC)
+  # if(BUILD_TESTING)
+  set(_NAME "opencensus_${NAME}")
+  add_executable(${_NAME} ${SRC})
+  prepend_opencensus(DEPS "${ARGN}")
+  target_link_libraries(${_NAME} "${DEPS}" benchmark)
+  # add_test(NAME ${_NAME} COMMAND ${_NAME}) endif()
```
Comment (g-easy) [inline cmake/OpenCensusHelpers.cmake]: Not sure what to do here, I'd be happy to drop the commented out line, and to not run benchmarks as part of CI.

File: cmake/OpenCensusDeps.cmake [hunk 4]
```diff
@@ -23,12 +23,17 @@ fetchcontent_declare(abseil
                      GIT_REPOSITORY
                      https://github.com/abseil/abseil-cpp
                      GIT_TAG
-                     master)
+                     2c8421e1c6cef0da9e8a20b01c15256ec9ec116d)
```
Comment (g-easy) [inline cmake/OpenCensusDeps.cmake]: Is this fixed by #307? Can this go back to master now?

**Comment by g-easy** [comment]
This is looking good!

File: cmake/OpenCensusDeps.cmake (hunk 4 above)
Comment (meastp) [reply cmake/OpenCensusDeps.cmake]: Unfortunately, no - I rebased on master and tested this now

File: cmake/OpenCensusHelpers.cmake (hunk 1 above)
Comment (meastp) [reply cmake/OpenCensusHelpers.cmake]: ok, fixed

File: cmake/OpenCensusHelpers.cmake (hunk 2 above)
Comment (meastp) [reply cmake/OpenCensusHelpers.cmake]: ok, fixed

File: cmake/OpenCensusHelpers.cmake (hunk 3 above)
Comment (meastp) [reply cmake/OpenCensusHelpers.cmake]: I tried to uncomment that line. I think benchmarks are useful to run during CI.

**Comment by meastp** [comment]
I have addressed your comments :)

File: cmake/OpenCensusDeps.cmake (hunk 4 above)
Comment (g-easy) [reply cmake/OpenCensusDeps.cmake]: Why does this have to be pinned? Is it to fix a build break? If so, please add a TODO.

File: cmake/OpenCensusHelpers.cmake (hunk 3 above)
Comment (g-easy) [reply cmake/OpenCensusHelpers.cmake]: I tried this on my workstation and the benchmarks are responsible for 53 out of 71 seconds of the `ctest` runtime. I would really prefer to not add this delay to CI.

File: cmake/OpenCensusDeps.cmake (hunk 4 above)
Comment (g-easy) [reply cmake/OpenCensusDeps.cmake]: I changed it back to master and it builds locally. Is ABSL [only] broken on Windows?

Comment (meastp) [reply cmake/OpenCensusDeps.cmake]: Yes, as I wrote, it's broken on my machine and I'm on Windows.

File: cmake/OpenCensusHelpers.cmake (hunk 3 above)
Comment (meastp) [reply cmake/OpenCensusHelpers.cmake]: ok, removed :)

**Comment by meastp** [comment]
ok, fixed comments :)

File: cmake/OpenCensusDeps.cmake (hunk 4 above)
Comment (g-easy) [reply cmake/OpenCensusDeps.cmake]: Sorry, I didn't realize it was Windows specific.

**Comment by g-easy** [comment]
Awesome! Thank you! Wanna go tick the benchmarks box in https://github.com/census-instrumentation/opencensus-cpp/issues/244 ? :)

**Pull request merged**